tests/hmm/hmumu_utils.py

```diff
@@ -1199,26 +1215,30 @@ def get_selected_muons(
     """
 
     if fsrphotons:
-        size = muons.numevents()
 
+        out_muons_fsrPhotonIdx = muons.fsrPhotonIdx
```

Comment:
this will just make a reference (not copy) to the original array, so you will still modify the original data. Can you change to `out_muons_fsrPhotonIdx  = NUMPY_LIB.array(muons.fsrPhotonIdx)` to make a copy? The idea is that you want functions to have no side effects, i.e. they should clearly only modify the outputs and not the original data.

Reply:
Done, thanks!